stablehlo/tests/interpret_probe.mlir

```diff
@@ -0,0 +1,67 @@
+// RUN: stablehlo-translate --interpret --probe-output-dir=%T -split-input-file %s
+
```

Comment:
Is it worth testing the numpy format? Maybe not, since the check dialect tests that its readable, but if we wanted to, could do something like:
```
// RUN: cat %T/probe0_iteration0.npy | FileCheck --check-prefix=CHECK-NUMPY
...
// CHECK-NUMPY: { some_format_to_match :  ... {{.*}} } 
```

Reply:
I think the roundtrip tests might be enough? Were you thinking of locking down the NumPY header (as the rest would be binary data) for V1 specifically?

Reply:
I think given that tests read the format and expect a value we should be covered.